bugsnag-plugin-android-ndk/src/main/jni/utils/serializer.c

```diff
@@ -417,12 +417,14 @@ void bsg_serialize_grouping_hash(const bugsnag_event *event, JSON_Object *event_
   }
 }
 
-void bsg_serialize_handled_state(const bugsnag_event *event, JSON_Object *event_obj) {
+void bsg_serialize_severity_reason(const bugsnag_event *event, JSON_Object *event_obj) {
   // FUTURE(dm): severityReason/unhandled attributes are currently
   // over-optimized for signal handling. in the future we may want to handle
   // C++ exceptions, etc as well.
   json_object_set_string(event_obj, "severity", bsg_severity_string(event->severity));
-  json_object_dotset_boolean(event_obj, "unhandled", event->unhandled);
+  bool unhandled = event->unhandled;
+  json_object_dotset_boolean(event_obj, "unhandled", unhandled);
+  json_object_dotset_boolean(event_obj, "severityReason.unhandledOverridden", !unhandled);
```

Comment:
Think it would be worth commenting here on why `!unhandled` is correct (due to lack of support of notify callbacks), as a helpful pointer in future. 

```suggestion
  json_object_dotset_boolean(event_obj, "severityReason.unhandledOverridden", !unhandled);
```

Reply:
Added some comments to the method